web/src/message_view_header.ts

```diff
@@ -149,7 +149,6 @@ function build_message_view_header(filter: Filter | undefined): void {
     // message_view_header on a template where it's never used
     if (filter && !filter.is_common_narrow()) {
         search.open_search_bar_and_close_narrow_description();
-        search.set_search_bar_text(narrow_state.search_string());
```

Comment:
This will already happen from `reset_searchbox` which called through `open_search_bar_and_close_narrow_description` (not in every case, but in the cases that we want).